clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -22,35 +20,33 @@ public void checkHubConnectionState() throws InterruptedException {
     }
 
     @Test
-    public void SendWithNoParamsTriggersOnHandler() throws InterruptedException {
-        AtomicInteger value = new AtomicInteger(0);
+    public void SendWithNoParamsTriggersOnHandler() throws Exception {
+        AtomicReference<Double> value = new AtomicReference<Double>(0.0);
```

Comment:
Curious why this was changed to an atomic double instead of being left as an integer.

Reply:
Gson actually has trouble deserializing directly into `Integer` type. It defaults to `Double`. [Random SO post about the issue](https://stackoverflow.com/questions/17090589/gson-deserialize-integers-as-integers-and-not-as-doubles). 
One work aroud that I don't like would be to check if the desired cast type is Integer and then do the needful to convert the Double into an int but that feels wrong. Going to file an issue for this and custom type deserialization as well.

Reply:
Ah, ok. Maybe throw a quick comment in there, but I'm OK with this given the explaination.